woocommerce/woocommerce-ios
# Application Password Login: Use site discovery to get root API endpoint

Part of #10924


## Description

Previously we were hardcoding the root endpoint for WP sites with `wp-json` suffix. This doesn't work for sites with pretty permalink disabled. This prevents such sites from logging in and using the app.

This PR fixes issue logging in with application password for such sites. Changes include:
- Added `SiteDiscoveryUseCase` to support site discovery. This use case will be reused for after login.
- Updated `WordPressSiteRemote` to use the new use case before hitting the root endpoint.
- Updated tests.

🗒️ In a subsequent PR, I'll handle updating the root endpoint for REST requests. For now, users still can't log in with site credentials or request any API.

## Testing instructions

- Prequisite: make sure that you have access to a self-hosted site without Jetpack.
- Disable pretty permalink for your site by navigating to wp-admin > Settings > Permalinks > select Plain for permalink structure.
- On the app, log out if needed.
- On the prologue screen, select Log In.
- Enter the address of your site and enter incorrect credentials.
- On the error alert, select log in with web view.
- Notice that the web view successfully displays the page for log in to your site.

## Screenshots

| Before | After |
| --- | --- |
| <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/47b7fc6a-c4a9-4c38-a15c-eb75bab6e006" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/7132da9f-bc2e-4f9f-b0e9-5b6f71e84454" width=320 />



---
